lib/Dialect/FIRRTL/Transforms/Dedup.cpp

```diff
@@ -183,10 +214,11 @@ struct StructuralHasher {
   unsigned currentIndex = 0;
   DenseMap<Value, unsigned> indexes;
 
-  // This is a set of every attribute we should ignore.
-  DenseSet<Attribute> nonessentialAttributes;
-  // This is a cached "portTypes" string attr.
-  StringAttr portTypesAttr;
+  // This keeps track of module names in the order of the appearance.
+  SmallVector<Attribute> referredModuleNames;
```

Comment:
I think this can be a `SmallVector<ArrayAttr>`, then `h->second.cast<ArrayAttr>().getAsRange<StringAttr>()` below simplifies.

Reply:
Done, thanks!